safe_transaction_service/history/admin.py

```diff
@@ -111,12 +112,26 @@ def block_number(self, obj: MultisigConfirmation) -> Optional[int]:
 
 @admin.register(ERC20Transfer)
 class ERC20TransferAdmin(TokenTransferAdmin):
-    pass
+    actions = ["to_erc721"]
+
+    @admin.action(description="Convert to ERC721 Transfer")
+    @atomic
+    def to_erc721(self, request, queryset):
+        for element in queryset:
+            element.to_erc721_transfer().save()
+        queryset.delete()
 
 
 @admin.register(ERC721Transfer)
 class ERC721TransferAdmin(TokenTransferAdmin):
-    pass
+    actions = ["to_erc20"]
+
+    @admin.action(description="Convert to ERC20 Transfer")
+    @atomic
+    def to_erc20(self, request, queryset):
+        for element in queryset:
+            element.to_erc20_transfer().save()
```

Comment:
Isn't there something like `build_update` that could be used so we don't necessarily iterate through everything? Maybe mapp everything and then bulk save to have a single DB access instead of one per item.

Reply:
The same could be done in line `119` of this file

Reply:
Not an easy way, this method should only be used for fixing some elements and not expecting a lot of transfers in the queryset